src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,65 @@
+package step2;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final Pattern STRING_PATTERN = Pattern.compile("//(.)\n(.*)");
+
+    public static int splitAndSum(String data) {
+        if (data == null || data.isBlank()) {
+            return 0;
+        }
+
+        String pattern = findPattern(data);
+        if (!pattern.isBlank()) {
+            String[] strings = getNumbers(data, pattern);
+            int result = 0;
+            for (String str : strings) {
+                int num = Integer.parseInt(str);
+                checkPositive(num);
+
+                result += num;
+            }
+
+            return result;
+        }
+
+        int aNumber = Integer.parseInt(data);
+        checkPositive(aNumber);
+
+        return aNumber;
+    }
+
+    private static String findPattern(String data) {
+        String delimiter = "";
+        Matcher m = STRING_PATTERN.matcher(data);
+        if (m.find()) {
+            return m.group(1);
+        }
+
+        try {
+            delimiter = data.split(",|:").length > 0 ? ",|:" : "";
+        } catch (Exception e) {
+            // nothing to do
+        }
+
+        return delimiter;
+    }
+
+    private static String[] getNumbers(String data, String pattern) {
+        Matcher m = STRING_PATTERN.matcher(data);
+        if (m.find()) {
+            return m.group(2).split(pattern);
+        }
+
+        return data.split(",|:");
+    }
```

Comment:
findPattern 메서드에서 커스텀 딜리미터를 찾으면 해당 딜리미터를 리턴하고
그렇지 않으면 기본 딜리미터를 반환하고 있습니다.
이후 결과를 다시 getNumbers 메서드에 보내서 문자열을 가져오는 방식이네요

```suggestion
    private static String findPattern(String data) {
        String delimiter = "";
        Matcher m = STRING_PATTERN.matcher(data);
        if (m.find()) {
            return m.group(1);
        }

        try {
            delimiter = data.split(",|:").length > 0 ? ",|:" : "";
        } catch (Exception e) {
            // nothing to do
        }

        return delimiter;
    }

    private static String[] getNumbers(String data, String pattern) {
        Matcher m = STRING_PATTERN.matcher(data);
        if (m.find()) {
            return m.group(2).split(pattern);
        }

        return data.split(pattern);
    }
```
그렇다면 getNumbers에서는 받은 pattern(구분자)을 기반으로 split해야하지 않을까요?
그리고 이 두 메서드를 합쳐보면 앞서말한 split이라는 메서드를 만들 수 있을 것 같습니다.

Reply:
명훈님은 pattern이라는 변수명을 사용하셨는데
pattern보다는 구분자 용도로 사용하고 있기에 delimiter라는 변수명이 더 적합해 보입니다.